cfme/tests/cloud/test_providers.py

```diff
@@ -94,6 +94,43 @@ def test_discovery_password_mismatch_validation_cloud(appliance):
     view.flash.assert_message('Password/Verify Password do not match')
 
 
+@pytest.mark.tier(3)
+@pytest.mark.uncollectif(lambda appliance: appliance.version >= '5.9',
+                        reason='no more support for cloud provider discovery')
+def test_discovery_error_azure_cloud(appliance):
+    """ Test Azure discovery with feck data
+
+    prerequisites:
+        * appliance supporting discovery
+
+    Steps:
+        * Navigate Cloud provider discovery and select Azure
+        * Fill all fields with feck data
+        * Start Discovery
+        * Even with wrong data discovery will start with the proper flash message assert it
+        * Check for provider should not discover
+    """
+    cred = Credential(
+        principal=fauxfactory.gen_alphanumeric(5),
+        secret=fauxfactory.gen_alphanumeric(8),
+        tenant_id=fauxfactory.gen_alphanumeric(10),
+        subscription_id=fauxfactory.gen_alphanumeric(10))
+
+    collection = appliance.collections.cloud_providers
+    view = navigate_to(collection, 'All')
+    initial_count = len(view.entities.entity_names)
+
+    collection.discover(cred, AzureProvider)
+    view = appliance.browser.create_view(CloudProvidersView)
+    view.flash.assert_success_message('Cloud Providers: Discovery successfully initiated')
+
+    # While waiting for new provider, TimeOutError will come (Negative Test)
+    with pytest.raises(TimedOutError):
+        collection.wait_for_new_provider()
```

Comment:
I think for this to be effective for use in tests like this, the method you're using needs to be updated to take a timeout kwarg, and pass it to the wait_for call. Currently it waits 1000s, or ~16 minutes, which is far too long for a test like this to just wait for an exception to occur that we're fully expecting.  Please add the required kwarg, and pass a much shorter timeout (maybe only 60s?) for this test. Currently it takes ~22m to run.

https://github.com/ManageIQ/integration_tests/blob/e4a5fa4e37f467f4e86784477ae596b948b58ab3/cfme/cloud/provider/__init__.py#L189

Secondly, is there ever another flash message for when discovery fails? Any further validation you could do that the discovery process has actually completed and failed and not just timed out?

Reply:
@mshriver I added a timeout. Now, it's taking 507 sec. Further validation not possible. I have to go for log check; no flash message for discovery fails.